pkg/artifactory/datasource_artifactory_file_test.go

```diff
@@ -1,58 +1,199 @@
+		  path            = "crash.zip"
+		  output_path     = "%s"
+		  force_overwrite = false
+		}
+	`
+	const forceOverWriteScript = `
+		data "artifactory_file" "example" {
+		  repository      = "example-repo-local"
+		  path            = "crash.zip"
+		  output_path     = "%s"
+		  force_overwrite = true
+		}
+	`
+
+	var skipDownloadCheck = func(state *terraform.State) error {
+		download := state.Modules[0].Resources["data.artifactory_file.example"].Primary.Attributes["output_path"]
+		downloadedFileStat, err := os.Stat(download)
+		if err != nil {
+			return err
+		}
+		downloadedFileModTime := downloadedFileStat.ModTime()
+		if downloadedFileModTime.After(localFileModTime) { //determines fresh download occurred during the test step
+			return fmt.Errorf("fresh download observed. Existing file modification time: %v. Fresh downloaded file modification time: %v", localFileModTime, downloadedFileModTime)
+		}
+		return nil
+	}
+
+	var downloadCheck = func(state *terraform.State) error {
+		download := state.Modules[0].Resources["data.artifactory_file.example"].Primary.Attributes["output_path"]
+		downloadedFileStat, err := os.Stat(download)
+		if err != nil {
+			return err
+		}
+		downloadedFileModTime := downloadedFileStat.ModTime()
+		verified, err := VerifySha256Checksum(download, "7a2489dd209d0acb72f7f11d171b418e65648b9cc96c6c351e00e22551fdd8f1")
+		if !verified {
+			return fmt.Errorf("%s checksum does not have expected checksum", download)
+		}
+		//makes sure download occurred.
+		if !downloadedFileModTime.After(localFileModTime) { //determines fresh download occurred during the test step
+			return fmt.Errorf("fresh download observed. Existing file modification time: %v. Fresh downloaded file modification time: %v", localFileModTime, downloadedFileModTime)
+		}
+		return err
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          downloadPreCheck(t, downloadPath, &localFileModTime),
 		ProviderFactories: testAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: script,
-				Check: func(state *terraform.State) error {
-					download := state.Modules[0].Resources["data.artifactory_file.example"].Primary.Attributes["output_path"]
-					_, err := os.Stat(download)
-					if err != nil {
-						return err
-					}
-					verified, err := VerifySha256Checksum(download, "7a2489dd209d0acb72f7f11d171b418e65648b9cc96c6c351e00e22551fdd8f1")
-
-					if !verified {
-						return fmt.Errorf("%s checksum does not have expected checksum", download)
-					}
-					return err
-				},
+				Config: fmt.Sprintf(noOverWriteForcedScript, downloadPath),
+				Check:  skipDownloadCheck,
+			},
+			{
+				Config: fmt.Sprintf(forceOverWriteScript, downloadPath),
+				Check:  downloadCheck,
 			},
 		},
 	})
 }
+
+//Creates new directory tree if not exist
+func createNewDir(srcPath string) error {
+	_, err := os.Stat(srcPath)
+	if err == nil {
+		return err // path exists, nothing to do
+	}
+	if !errors.Is(err, os.ErrNotExist) {
```

Comment:
What happens if you remove lines 57,58,59?

You'd pass `nil` as `err` to this function. If it blows up, well we know it's needed. If it doesn't, then 

```
if true
   return err. 
```
So, if error is nil, and this returns `true` (which is to say `!ErrNotExist`), then we return nil. Good. If it's not nil, and isn't `ErrorNotExist`, then return whatever err it was. Also good.

Please experiment and literally remove lines 157-159

Reply:
code improved.